clients/src/main/java/org/apache/kafka/common/Endpoint.java

```diff
@@ -16,35 +16,51 @@
  */
 package org.apache.kafka.common;
 
-import org.apache.kafka.common.annotation.InterfaceStability;
 import org.apache.kafka.common.security.auth.SecurityProtocol;
 
+import java.util.Locale;
 import java.util.Objects;
 import java.util.Optional;
 
 /**
  * Represents a broker endpoint.
  */
 
-@InterfaceStability.Evolving
 public class Endpoint {
 
     private final String listenerName;
     private final SecurityProtocol securityProtocol;
     private final String host;
     private final int port;
 
+    public static String parseListenerName(String connectionString) {
```

Comment:
it seems only `KafkaConfig` uses this helper. Could you please move it to the `KafkaConfig` instead of leaving it in this public APIs?